src/EventListener/BeforeRegistrationStartsListener.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopware\AppBundle\EventListener;
+
+use Shopware\App\SDK\Event\BeforeRegistrationStartsEvent;
+use Shopware\AppBundle\Exception\ShopURLIsNotReachableException;
+use Symfony\Component\EventDispatcher\Attribute\AsEventListener;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+#[AsEventListener]
+class BeforeRegistrationStartsListener
+{
+    private HttpClientInterface $httpClient;
+
+    private bool $checkShopURLIsReachable;
+
+    public function __construct(
+        HttpClientInterface $httpClient,
+        bool $checkShopURLIsReachable
+    ) {
+        $this->httpClient = $httpClient;
+        $this->checkShopURLIsReachable = $checkShopURLIsReachable;
+    }
+
+    public function __invoke(BeforeRegistrationStartsEvent $event): void
+    {
+        if ($this->checkShopURLIsReachable === false) {
+            return;
+        }
+
+        $shop = $event->getShop();
+
+        try {
+            $this->httpClient->request('HEAD', $shop->getShopUrl(), [
```

Comment:
I would really do an Admin API call to get the version (that requires no version). The benefit is then that you know that the shop you are connecting back to is the actual one

Sometimes people copy their production to staging and don't change the APP_URL, therefore

Reply:
As agreed we could keep this change here